data_structures/hashing/bloom_filter.py

```diff
@@ -0,0 +1,105 @@
+"""
+See https://en.wikipedia.org/wiki/Bloom_filter
+"""
+from hashlib import md5, sha256
+from random import choices
+from string import ascii_lowercase
+
+
+class Bloom:
+    # number of hash functions is fixed
+    HASH_FUNCTIONS = (sha256, md5)
+
+    def __init__(self, size: int = 8) -> None:
+        self.bitstring = 0b0
+        self.size = size
+
+    def add(self, value: str) -> None:
+        h = self.hash_(value)
+        self.bitstring |= h
+        print(
+            f"""\
+[add] value =      {value}
+      hash =       {self.format_bin(h)}
+      filter =     {self.format_bin(self.bitstring)}
+"""
+        )
+
+    def exists(self, value: str) -> bool:
+        h = self.hash_(value)
+        res = (h & self.bitstring) == h
+
+        print(
+            f"""\
+[exists] value =   {value}
+         hash =    {self.format_bin(h)}
+         filter =  {self.format_bin(self.bitstring)}
+         res =     {res}
+"""
+        )
```

Comment:
```suggestion
```
In the `CONTRIBUTING` it says
> - return all calculation results instead of printing or plotting them

I don't think these prints are necessary

Reply:
moved to method and called in doctest